testtools/matchers/_const.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2016 testtools developers. See LICENSE for details.
+
+__all__ = [
+    'always',
+    'never',
+    ]
+
+from testtools.compat import _u
+from ._impl import Mismatch
+
+
+class _Always(object):
+    """Always matches."""
+
+    def match(self, value):
+        return None
+
+
+def always():
+    """Always match."""
+    return _Always()
+
+
+class _Never(object):
+    """Never matches."""
+
+    def match(self, value):
+        return Mismatch(
+            _u('Inevitable mismatch on %r' % (value,)))
+
+
+def never():
```

Comment:
In my head, I was thinking this would take an optional mismatch to return. Unfortunately, there's no nice way to parametrize it against the value being matched. OK to leave as-is for now. We can extend the interface later if we need to.